compiler/rustc_hir_typeck/src/callee.rs

```diff
@@ -712,6 +713,119 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         err.emit();
     }
 
+    /// Enforces that functions being called actually have the effects that the current context
+    /// requires for them being callable.
+    pub(super) fn enforce_context_effects(
+        &self,
+        hir_id: hir::HirId,
+        span: Span,
+        value: Ty<'tcx>,
+    ) -> Ty<'tcx> {
```

Comment:
does this function always just return `value`?

Reply:
also, can this be replaced by emitting a `value: const<Env> FnOnce(?args) -> ?ret` bound?